src/main/firebase/FirebaseController.ts

```
@@ -155,6 +170,65 @@ export class FirebaseDatabase {
+            }
+        } catch (error) {
+            console.error("error for user " + userId);
+            console.error(error);
+            return Promise.reject(error);
+        }
+        return Promise.resolve();
+    }
+
+    async  disableSourcesMonitoringAnValidationAndDisableUserStatus(userId: string, sources: any) {
+        let sourcePromises: any = [];
+        let sourcesToDisablePromises: any = [];
+        let disabledSourcesId: string[] = [];
+        try {
+            if (sources) {
+                for (const key in sources) {
+                    if (sources.hasOwnProperty(key) && sources[key] === "owner") {
+                        sourcePromises.push(this.getSource({ id: key, secretKey: "" }));
+                    }
+                }
+                const sourcesResutlt: any = await Promise.all(sourcePromises);
+                for (const result of sourcesResutlt) {
+                    const source = result.val();
+
+                    if (source.validation_enabled) {
+                        sourcesToDisablePromises.push(
+                            this.updateSource(source.id, {
+                                validation_enabled: false, monitoring_enabled: false,
+                                maxAverageResponseTime: {}, maxErrors: {}
+                            })
+                        );
+                        disabledSourcesId.push(source.id);
+                    }
+                }
+            }
+            sourcesToDisablePromises.push(
+                this.updateUser(userId, { disabled_sources: disabledSourcesId, status: userStatus.Disabled })
+            );
+
+            await Promise.all(sourcesToDisablePromises);
+            return Promise.resolve();
+        } catch (error) {
+            console.error("There was a problem trying to deactivate source monitoring or update user status, userID : ", userId);
+            return Promise.reject(error);
```

throw error instead of return Promise.reject